{bio}[foss/2020b} pangolin v3.1.10

sassy-crick opened the pull request:
(created using `eb --new-pr`)

edit: requires ~#13708~ (UShER), ~#13651~ (gofasta)

Micket commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,75 @@
+        'checksums': ['48390ed1c1a79bb5f801083042cdeddee060e2277f40c293fd30384882aced81'],
+    }),
+    ('constellations', '0.0.13', {
+        'source_urls': ['https://github.com/%(github_account)s/%name)s/archive'],
```
```suggestion
        'source_urls': ['https://github.com/%(github_account)s/%(name)s/archive'],
```

sassy-crick replied:
As below.

Micket commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,75 @@
+    }),
+    ('constellations', '0.0.13', {
+        'source_urls': ['https://github.com/%(github_account)s/%name)s/archive'],
+        'sources': [{'download_filename': 'v%(version).tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
```
```suggestion
        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
```

sassy-crick replied:
Thanks for finding them! I simply did not spot them. (I do need new glasses, not a joke). I have corrected them.
By the way, is there  a way to avoid these repeating lines?

sassy-crick replied:
Converted to draft as we are waiting to have `UShER` merged.

Flamefire replied:
> By the way, is there a way to avoid these repeating lines?

You can likely put them into `exts_default_options` (dict that is merged with this dict) although I'm not fully sure it works with the templates, but please try.
Also use `SOURCE_TAR_GZ` as the `filename` and `GITHUB_SOURCE` as the source_url, see https://docs.easybuild.io/en/latest/version-specific/easyconfig_templates.html

> Converted to draft as we are waiting to have UShER merged.

If that is the only thing then simply mention it in the PR description like: `- [ ] Requires #13708`

Flamefire commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,75 @@
+}
+
+sanity_check_commands = [
+    'pangolin -pv && scorpio -v'
```
Minor: 
```suggestion
    'pangolin -pv',
    'scorpio -v',
```

This way they are checked independently and if one errors the message is a bit better.

sassy-crick replied:
Agreed and changed.

Flamefire commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,75 @@
+    }),
+]
+
+_binaries = [
```
Use `local_binaries` to pass CI checks

sassy-crick replied:
Agreed and changed. I will try to remember to use `local_XX` in future.

Flamefire commented:
The dependencies and extensions seem a bit off. I guess you used https://github.com/cov-lineages/pangolin/blob/master/environment.yml as a reference?
Checking https://github.com/cov-lineages/pangolin/blob/504d9833f4715c26b08b30e1cd2ad859d358c721/setup.py#L14-L22 there seem to be a few Python packages missing. E.g. pandas. Can you double check those?

SethosII commented:
@Flamefire `pandas` for example is included in `SciPy-bundle`

sassy-crick commented:
> The dependencies and extensions seem a bit off. I guess you used https://github.com/cov-lineages/pangolin/blob/master/environment.yml as a reference?
Yes, that is what I done and also added what was missing when I installed it. This is where the `scikit-learn` for example came from. 
@SethosII has already addressed the issue with `pandas` I will look into the other two I noticed `joblib` and `PuLP` and see if they are really missing or being pulled in. 
I noticed that for `scikit-learn` `SciPy-bundle` is a requirement so we could remove that one.

Flamefire commented:
>  I noticed that for scikit-learn SciPy-bundle is a requirement so we could remove that one.

Keep it. It is better to have the direct dependencies explicit in case the others change in a later version

sassy-crick commented:
> The dependencies and extensions seem a bit off. I guess you used https://github.com/cov-lineages/pangolin/blob/master/environment.yml as a reference?
Yes, that is what I done and also added what was missing when I installed it. This is where the `scikit-learn` for example came from. 

@SethosII has already addressed the issue with `pandas`. I have checked this [link][https://github.com/cov-lineages/pangolin/blob/504d9833f4715c26b08b30e1cd2ad859d358c721/setup.py#L14-L22) and compared it with what I got:
```biopython                     1.78 (>=1.70)
pandas                        1.1.4  (>=1.0.1)
wheel                         0.35.1 (>=0.34)
joblib                        0.17.0 (>=0.11)
pysam                         0.16.0.1 (>1.16.0)
scikit-learn                  0.23.2 (>= 0.23.1)
PuLP                          2.4 ((>=2)
```
So that should be ok.

I looked into getting the repeating lines in the extension list a bit more neat. I replaced the URLs with the GITHUB_SOURCE as suggested. However, the SOURCE_TAR_GZ will only work in one case as all the others are named **v**-name-version.tar.gz. I tried to put the checksums outside the `exts_list` but here the style-check failed.

sassy-crick commented:
Additionally, to move forward, added Kent_tools so I can removed them from [UShER](https://github.com/easybuilders/easybuild-easyconfigs/pull/13694)

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3582.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/7821359089b03a595ecfa5f248427bae for a full test report.

boegel commented:
@boegelbot please test @ generoso

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2609.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/77c558db9afc772baa7b54fce897ade2 for a full test report.

SethosII commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,80 @@
+# Contribution from the NIHR Biomedical Research Centre
+# Guy's and St Thomas' NHS Foundation Trust and King's College London
+# Based on a EC provided by Paul Jähn
```
Just a nitpick: my last name has an `e` at the end ;)

```suggestion
# Based on a EC provided by Paul Jähne
```

SethosII commented on easybuild/easyconfigs/p/pangolin/pangolin-3.1.10-foss-2020b.eb:
```diff
@@ -0,0 +1,80 @@
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_commands = [
```
I would also add `pangolin -v`:

```suggestion
sanity_check_commands = [
    'pangolin -v',
```

boegel reviewed (approved):
lgtm

boegel commented:
Going in, thanks @sassy-crick!
